envbuilder.go

```diff
@@ -156,6 +157,34 @@ func Run(ctx context.Context, options Options) error {
 		}
 	}
 
+	gitURLParsed, err := url.Parse(options.GitURL)
+	if err != nil {
+		return fmt.Errorf("invalid git URL: %w", err)
+	}
+	// If we're cloning over SSH, we need a known_hosts file.
+	if gitURLParsed.Scheme == "ssh" {
```

Comment:
I believe it's possible to clone over SSH even without `ssh://`. Just `git@github.com:org/repo` should suffice. I wonder if this will catch that case?

Reply:
It will actually refuse to accept that URL, as git URLs are not all valid URLs. 
https://go.dev/play/p/sXvpyxwNLIq

To handle these sorts of URLs, I suppose we could treat all urls without schema prefixes as `ssh://`. WDYT?

Reply:
I think it is valid since they don't follow any technical convention format.